PSICompile/ILCodeGen.cs

```diff
@@ -106,22 +174,12 @@ public class ILCodeGen : Visitor {
       });
    }
 
-   public override void Visit (NIdentifier d) {
-      switch (mSymbols.Find (d.Name)) {
-         case NConstDecl cd: Visit (cd.Value); break;
-         case NVarDecl vd:
-            var type = TMap[vd.Type];
-            if (vd.Local) Out ($"    ldloc {vd.Name}");
-            else Out ($"    ldsfld {type} Program::{vd.Name}");
-            break;
-         default: throw new NotImplementedException ();
-      }
-   }
+   public override void Visit (NIdentifier d) => LoadIdent (d.Name);
 
    public override void Visit (NUnary u) {
       u.Expr.Accept (this);
       string op = u.Op.Kind.ToString ().ToLower ();
-      op = op switch { "sub" => "neg", _ => op };
+      op = op switch { "sub" => "neg", "not" => "ldc.i4.0\n    ceq", _ => op };

```

Comment:
not should work for boolean as well as integer. 
`WriteLn (not true)` should output **false**.
`WriteLn (not 2)` should output **-3** (which is the value you get when you flip all the bits in the binary representation of 2). You should use simply the **not** IL opcode for that situation. So the code you generate for not should depend on the type of the underlying expression.